debian/control

```diff
@@ -0,0 +1,11 @@
+Source: movai-ignition-sensors6
+Maintainer: Mariana Alves<mariana.alves@mov.ai>
+Section: unknown
+Priority: optional
+Build-Depends: unzip, debhelper (>= 7.4.3)
+Standards-Version: 3.9.1
+
+Package: movai-ignition-sensors6
+Architecture: all
+Depends: ${misc:Depends}
```

Comment:
missing the runtime dependencies :
```
libc6 (>= 2.29), libgcc-s1 (>= 3.0), libignition-common4 (>= 4.7.0), libignition-math6 (>= 6.14.0), libignition-msgs8 (>= 8.7.0), libignition-transport11 (>= 11.4.1), libprotobuf17, libsdformat12 (>= 12.7.1), libstdc++6 (>= 5.2)
```

Reply:
maybe this can help:
The `Depends` field in the Debian control file is typically populated manually by the package maintainer, and it specifies the package dependencies needed for the software to run. This information is usually not automatically extracted from the `CMakeLists.txt` file or any other build-related files.

However, you can automate the process to some extent by integrating tools or scripts into your build process. For example, you might create a script that analyzes your project's build system (like `CMakeLists.txt`) and extracts the dependencies. This script could then generate the appropriate `Depends` field for the Debian control file.

As for passing dependencies as an argument to `dpkg-buildpackage`, the typical approach is to specify the dependencies in the `debian/control` file. While you can pass arguments to the build process, it's not a common practice to dynamically generate the `Depends` field during the build.

Here's a simplified example of how you might use a script to generate the `Depends` field based on your project's dependencies:

1. Create a script (e.g., `generate-depends.sh`) that extracts dependencies from your project.

```bash
#!/bin/bash

# Replace this with logic to extract dependencies from your build system
# For simplicity, let's assume you have a file called "dependencies.txt"
DEPENDENCIES=$(cat dependencies.txt)

echo "Depends: ${DEPENDENCIES}"
```

2. In your `debian/control` file, reference the generated `Depends` field.

```plaintext
Source: your-source-package
Section: devel
Priority: optional
Maintainer: Your Name <your.email@example.com>
Build-Depends: debhelper (>= 12)
Standards-Version: 4.5.0
Homepage: http://www.example.com

Package: your-package
Architecture: any
Depends: ${sh:Depends}, ${misc:Depends}, ${generated:Depends}
Description: Your package description
```

3. Modify your build process to include the script execution.

```bash
#!/bin/bash

# Other build steps

./generate-depends.sh > debian/generated-depends
dpkg-buildpackage -us -uc -rfakeroot

# Clean up
rm debian/generated-depends
```

This is just a basic example, and you may need to adjust it based on your project's specific requirements.

Reply:
The list of dependencies you provided do not match the information provided by the cmakelists file (for example the libignition-cmake is not in the list, but is a requirement in the CMakeLists.txt), so I don't think there is a dynamic way to retrieve them, at least in a way that will match the list provided in the package generated manually thorugh the ignition process (jenkins).